fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/qrcode/QrScannerFragment.java

```diff
@@ -0,0 +1,227 @@
+package com.github.dedis.popstellar.ui.qrcode;
+
+import android.Manifest;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.*;
+
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.camera.mlkit.vision.MlKitAnalyzer;
+import androidx.camera.view.LifecycleCameraController;
+import androidx.core.content.ContextCompat;
+import androidx.fragment.app.Fragment;
+
+import com.github.dedis.popstellar.databinding.QrScannerFragmentBinding;
+import com.github.dedis.popstellar.utility.Constants;
+import com.google.mlkit.vision.barcode.*;
+import com.google.mlkit.vision.barcode.common.Barcode;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.Executor;
+
+import static android.content.pm.PackageManager.PERMISSION_GRANTED;
+import static androidx.camera.view.CameraController.COORDINATE_SYSTEM_VIEW_REFERENCED;
+import static androidx.core.content.ContextCompat.checkSelfPermission;
+
+public class QrScannerFragment extends Fragment {
+  public static final String TAG = QrScannerFragment.class.getSimpleName();
+
+  public static final String SCANNING_KEY = "scanning_action_key";
+
+  private QrScannerFragmentBinding binding;
+  private BarcodeScanner barcodeScanner;
+
+  private QRCodeScanningViewModel viewModel;
+
+  public QrScannerFragment() {
+    // Required empty public constructor
+  }
+
+  /**
+   * This is for scan other than roll call
+   *
+   * @param scanningAction which scanning action is to be performed
+   * @return a QrScannerFragment with correct arguments in bundle
+   */
+  public static QrScannerFragment newInstance(ScanningAction scanningAction) {
+    QrScannerFragment fragment = new QrScannerFragment();
+    Bundle bundle = new Bundle(1);
+    bundle.putSerializable(SCANNING_KEY, scanningAction);
+    fragment.setArguments(bundle);
+    return fragment;
+  }
+
+  /**
+   * This newInstance is for RollCall scan of attendees only and must be used.
+   *
+   * @param rcPersistentId the id of the roll call
+   * @return a QrScannerFragment with correct arguments in bundle
+   */
+  public static QrScannerFragment newRollCallScanInstance(String rcPersistentId) {
+    QrScannerFragment fragment = new QrScannerFragment();
+    Bundle bundle = new Bundle(2);
+
+    // We know that this newInstance if for roll call
+    bundle.putSerializable(SCANNING_KEY, ScanningAction.ADD_ROLL_CALL_ATTENDEE);
+    bundle.putString(Constants.ROLL_CALL_ID, rcPersistentId);
+    fragment.setArguments(bundle);
+    return fragment;
+  }
```

Comment:
Just in case there is a confusion it's string (singular) the `ROLL_CALL_ID` is the argument key (which is then used to retrieve tha value).
I also dislike having that stuff in the fragment because the scanner should be agnostic to roll call specifics. But since it's needed, I personally prefer separating the newInstance methods. That way one of them may remain unpolluted and the why of the string argument handling seems clearer to me so. 
Nevetheless, if you have strong feeling about it, I can change it. I'd also be intersted to have @GabrielFleischer's opinion on the matter

Reply:
If I am not mistaken, the argument is only used for the back callback.

What could be done is to rename `ROLL_CALL_ID` to `BACK_ARGS` which would then become a vararg that is passed to the backPressedCallback

That way there would be only one factory function, and it would be independent of the roll call feature.

Reply:
Instead of having a very specific use case for the roll call scanner, what we can do is to generalize the backPressedCallBack argument to any argument given in the bundle.

So, the changes would be :

- In the arguments of the fragment, the `ROLLCALL_ID` becomes `BACK_PRESSED_ARGS` and stores an array of string
- `onBackPressedCallback` now takes a (list, array, vararg ?) argument instead of a string as second argument
- When producing the `onBackPressedCallback`, you always use the  `BACK_PRESSED_ARGS`

Reply:
Oh I see now, it's much better. Thank you both for your partience with me 🙂